pentaho-js/src/main/javascript/cdf/components/VisualizationAPIComponent.js

```diff
@@ -25,24 +30,49 @@ define([
 
     render: function(data) {
       var domElem = this.placeholder()[0];
-      var wrapper = new VisualWrapper(domElem);
-      wrapper.data = data;
-      wrapper.visualSpec = this.getVisualSpec();
-      wrapper.update()
-        .then(_.bind(this.endExec, this), _.bind(this.failExec, this));
-    },
-  	
-    getVisualSpec: function() {
-      var visualSpec = {};
-      
-      _.each(this.vizOptions, function(option) {
-        visualSpec[option[0]] = this.getParameterValue(option[1]);
-      }, this.dashboard);
+      var contextVars = new GlobalContextVars({application: "pentaho-cdf"});
+
+      var _context = new Context(contextVars);
+      var me = this;
+      _context.getAsync(this.vizId)
+          .then( function (Model) {
+            var createdModel = new Model({});
+            createdModel.data = new Table(data);
+            createdModel.height= me.height;
+            createdModel.width = me.width;
+
+            $.each(me.vizOptions, function(i, v) {
+              var prop = createdModel.get(v[0]);
```

Comment:
This is not the "property type" instance, but the _current value_ of the property. If it is `null`, the line further down: `prop.type`, would throw... When it does not throw the obtained type is the type of the stored value, which can be a subtype of the declared property type.

To get the "property type" instance:

```js
var propType = createdModel.type.get(v[0]);

// The type of the value stored by properties of this type:
if(propType.type.isSimple) {
}
```